### DISCUSSION TITLE
Adding visual signature appearance from image file

### DISCUSSION THREAD

**Discussion by agunghp** [gettalong/hexapdf#281] [Q&A] [answered]
Hi,

Is there any code example to add a visual appearance of a signature from an image file?

I have tried with the code below but got an error message from Adobe Reader "Expected a dict object".
I'm not sure if it's caused by the code below or a separate issue.

```
. . .
image_signature = doc.images.add('/path/to/image.png')
sig_field = doc.acro_form(create: true).create_signature_field('signature')
widget = sig_field.create_widget(doc.pages[0], Rect: [20, 20, 120, 120])
widget.create_appearance
      .canvas.xobject(image_signature, at: [0, 0])
. . .
```

Thanks for the help

**Comment by gettalong** [comment, marked as answer]
This should work fine. There was a Adobe problem (see https://github.com/gettalong/hexapdf/issues/251) but that was worked around with newer versions of HexaPDF.

Which version of HexaPDF did you use?

**Reply by agunghp** [reply]
It works! with this magic line 😄
```ruby
image[:Resources] = {}
``` 
We are using version 0.29.0.
So, in the latest version, that line is no longer needed?

**Reply by gettalong** [reply]
Yeah, it was fixed in 0.33.0. Upgrading is strongly recommended!

**Reply by agunghp** [reply]
Well noted. Thank you very much! 🙏
